blocks/table/src/components/table/table.tsx

```diff
@@ -153,41 +170,104 @@ export const Table = ({ blockEntity, updateEntity, readonly }: TableProps) => {
         setHeaderMenu({ col, bounds });
       }, 100);
     });
-  };
+  }, [localColumns, updateStateAndEntity]);
 
-  const addNewRow = () => {
-    const newLocalRows = [...rows, { rowId: `${rows.length}` }];
+  const handleAddNewRow = useCallback(() => {
+    const newLocalRows = [...localRows, {}];
     updateStateAndEntity({ newLocalRows });
-  };
+  }, [localRows, updateStateAndEntity]);
+
+  const handleCellsEdited = useCallback<
+    NonNullable<DataEditorProps["onCellsEdited"]>
+  >(
+    (newValues) => {
+      const mergedLocalRows =
+        numberOfRows > localRows.length
+          ? [
+              ...localRows,
+              ...Array<Object>(numberOfRows - localRows.length).fill(
+                {} as Object,
```

Comment:
This seems to pass linting and saves the assertion.

The `Object` that `localRows` is an array of is actually a type named `Object` exported from `@blockprotocol/graph` (I think this is undesirable as it clashes with a JS primitive, and created H-1170 to rename it). It is an alias for `{}`.

I think the type would be better as a JsonObject, e.g. something with string keys and unknown json-permitted values. That requires changing the BP package though and would have knock-on consequences.
```suggestion
              ...Array<{}>(numberOfRows - localRows.length).fill({}),
```

Reply:
Thanks for the diff, have applied it in https://github.com/hashintel/hash/pull/3838/commits/dc8dbd2bef2f5f1d45633e272cfd5af446c6bcf2. I don't like that we're using `Object` here rather than properly defining nested properties, but didn't want to tackle this now as we likely will be changing the table block's definition further once we've agreed on what changes we're making in next week's sprint. Probably will get rid of this then.